File: src/SkyBlock/enchants/sword/DeathBringer.php
<?php


namespace SkyBlock\enchants\sword;

use pocketmine\event\entity\EntityDamageEvent;
use pocketmine\player\Player;
use SkyBlock\enchants\BaseEnchantment;

class DeathBringer extends BaseMeleeEnchant {

    static int $id = 102;

    public function isApplicableTo(Player $holder) : bool {
        $heldItem = $holder->getInventory()->getItemInHand();
        $ench = $heldItem->getEnchantment(BaseEnchantment::getEnchantment(self::$id));
        if ($ench !== null) {
            $level = $ench->getLevel();
            if ($level === 15) {
                return mt_rand(1, 20) === 1;
            } elseif ($level === 14) {
                return mt_rand(1, 21) === 1;
            } elseif ($level === 13) {
                return mt_rand(1, 22) === 1;
            } elseif ($level === 12) {
                return mt_rand(1, 23) === 1;
            } elseif ($level === 11) {
                return mt_rand(1, 24) === 1;
            } else {
                return mt_rand(1, 25) === 1;
            }
        }
        return false;
    }

    public function onActivation(Player $player, Player $attacker, EntityDamageEvent $ev, int $enchantmentlevel) : void {
        $ev->setModifier($enchantmentlevel > 10 ? $ev->getBaseDamage() : $ev->getBaseDamage() * ($enchantmentlevel / 10), 102);
        $this->sendActivation($attacker, "§cStruck by §bDeathBringer §cEnchant!");
        $this->sendActivation($player, "§bDeathBringer §aActivated!");
    }

}

File: SkyblockCore/src/main/java/dev/skyblock/enchants/sword/DeathBringer.java
package dev.skyblock.enchants.sword;
import dev.skyblock.enchants.BaseMeleeEnchant;
import org.bukkit.entity.Player;
import org.bukkit.event.entity.EntityDamageByEntityEvent;
public class DeathBringer extends BaseMeleeEnchant {
    public static final int ID = 102;
    public DeathBringer() { super(ID, "DeathBringer", 15); }
    @Override
    public boolean isApplicableTo(Player holder, int level) { return java.util.concurrent.ThreadLocalRandom.current().nextInt(1, 26) == 1; }
    @Override
    public void onActivation(Player victim, Player attacker, EntityDamageByEntityEvent ev, int level) {
        if (victim.getHealth() < 6) {
            attacker.sendMessage("§bDeathBringer §aActivated!");
            ev.setDamage(ev.getDamage() * 2);
        }
    }
}
